ReactiveCocoa/Swift/SignalProducer.swift

```diff
@@ -262,23 +261,39 @@ public struct SignalProducer<Value, Error: ErrorType> {
 	}
 }
 
+/// A uniquely identifying token for Observers that are replaying values in
+/// BufferState.
+private final class ReplayBuffer<Value> {
+	private var values: [Value] = []
+}
+
+
 private struct BufferState<Value, Error: ErrorType> {
-	// All values in the buffer.
+	/// All values in the buffer.
 	var values: [Value] = []
 
-	// Any terminating event sent to the buffer.
-	//
-	// This will be nil if termination has not occurred.
+	/// Any terminating event sent to the buffer.
+	///
+	/// This will be nil if termination has not occurred.
 	var terminationEvent: Event<Value, Error>?
 
-	// The observers currently attached to the buffered producer, or nil if the
-	// producer was terminated.
+	/// The observers currently attached to the buffered producer, or nil if the
+	/// producer was terminated.
 	var observers: Bag<Signal<Value, Error>.Observer>? = Bag()
+	
+	/// The set of unused replay token identifiers.
+	var replayBuffers: Bag<ReplayBuffer<Value>>? = Bag()
```

Comment:
This could be `Bag<ReplayBuffer<Value>>` since you don't set `nil` to this property.